fv3config/_asset_list.py

```diff
@@ -221,28 +214,25 @@ def config_to_asset_list(config):
         if is_dict_or_list(config['patch_files']):
             asset_list += ensure_is_list(config['patch_files'])
         else:
-            raise ConfigError('patch_files item in config dictionary must be a dict or list')
+            raise ConfigError(
+                'patch_files item in config dictionary must be an asset dict or '
+                'list of asset dicts')
     return asset_list
 
 
 def link_file(source_item, target_item):
+    if is_gsbucket_url(source_item) or is_gsbucket_url(target_item):
+        raise NotImplementedError(
+            'cannot perform linking operation involving remote urls '
+            f'from {source_item} to {target_item}'
+        )
     if os.path.exists(target_item):
         os.remove(target_item)
     os.link(source_item, target_item)
 
 
 def copy_file(source_path, target_path):
     if is_gsbucket_url(source_path):
```

Comment:
Same trick here as I describe above to create the filesystem object in a general fashion.